src/jokes.ts

```diff
@@ -60,6 +60,41 @@ class JokesLoader {
     }
   }
 
+  public async deleteJoke(joke: Joke): Promise<{ success: boolean; joke_id?: number; error?: string }> {
+    const jokesPath = path.join(__dirname, '../blagues.json');
+    const req = await this.checkAccess(jokesPath);
+
+    if (!req.success) return req;
+
+    try {
+      await this.loader.wait();
+
+      const rawData = await fs.readFile(jokesPath, 'utf-8');
+      const jokes = (rawData.length ? JSON.parse(rawData) : []) as Joke[];
+      const toDelJoke = jokes.find((j) => j.id === joke.id)!;
+      const index = jokes.indexOf(toDelJoke);
+
+      for (const joke of jokes) {
+        const jIndex = jokes.findIndex((j) => j === joke);
+        joke.id -= jIndex > index ? 1 : 0;
```

Comment:
Tu peut direct le mettre sur une seul ligne :)
```suggestion
        joke.id -= jokes.findIndex((j) => j === joke) > index ? 1 : 0;
```